moped-etl/data-tracker-sync/data_tracker_sync.py

```diff
@@ -0,0 +1,198 @@
+
+    updated = app.record(
+        method="update",
+        data=knack_project_record,
+        obj=KNACK_DATA_TRACKER_PROJECT_OBJECT,
+    )
+
+    knack_record_id = updated["id"]
+    return knack_record_id
+
+
+def main(args):
+    app = knackpy.App(
+        app_id=KNACK_DATA_TRACKER_APP_ID, api_key=KNACK_DATA_TRACKER_API_KEY
+    )
+
+    # Find all projects that are not synced to Data Tracker
+    unsynced_moped_projects = find_unsynced_moped_projects(is_test=args.test)
+
+    # Create a Knack project for each unsynced Moped project
+    created_knack_records = []
+    for project in unsynced_moped_projects:
+        moped_project_id = project["project_id"]
+        knack_record_id = create_knack_project_from_moped_project(
+            app=app, moped_project_record=project, is_test=args.test
+        )
+        created_knack_records.append(
+            {
+                "moped_project_id": moped_project_id,
+                "knack_record_id": knack_record_id,
+            }
+        )
+
+        # Update Moped project with Knack record ID of created record
+        update_moped_project_knack_id(moped_project_id, knack_record_id)
+
+    # Find all projects that are synced to Data Tracker to update them
+    synced_moped_projects = find_synced_moped_projects(
+        last_run_date=args.date, is_test=args.test
+    )
+
+    # Update synced Moped projects in Data Tracker and skip those just created
+    updated_knack_records = []
+    updates_to_skip = [record["moped_project_id"] for record in created_knack_records]
+
+    for project in synced_moped_projects:
+        if project["project_id"] in updates_to_skip:
+            continue
+
+        moped_project_id = project["project_id"]
+        knack_record_id = update_knack_project_from_moped_project(
+            app=app, moped_project_record=project, is_test=args.test
+        )
+        updated_knack_records.append(
+            {"moped_project_id": moped_project_id, "knack_record_id": knack_record_id}
+        )
+
+    logger.info(f"Done syncing.")
+    logger.info(f"Created {len(created_knack_records)} new Knack records")
+    logger.debug(f"Records created: {created_knack_records}")
+    logger.info(f"Updated {len(updated_knack_records)} existing Knack records")
+    logger.debug(f"Updated Knack records: {updated_knack_records}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-d",
+        "--date",
+        type=str,
+        default=datetime.now(timezone.utc).isoformat(),
+        help=f"ISO date string (in UTC) of latest updated_at value to find project records to update.",
+    )
+
+    parser.add_argument("-t", "--test", action="store_true")
+
+    args = parser.parse_args()
+
+    log_level = logging.DEBUG if args.test else logging.INFO
+    logger = get_logger(name="moped-knack-sync", level=log_level)
+    logger.info(
+        f"Starting sync. Will update records updated in Moped since {args.date}"
```

Comment:
I'm not sold on this phrasing, but I cant come up with anything better. 

Updates records in Data Tracker and Moped that have changed since last update? thats not much better....

Reply:
You and me both @chiaberry! I snuck this in last minute to log the arg coming out of Airflow. I'll update to: 
**Starting sync. Creating Knack records for Moped projects not synced and updating synced Knack records with latest project data from Moped.**
I think that is getting there. What do you think?

Reply:
awesome - thanks for the feedback! it inspired me to get proper docs in these function too. 🙏